### ISSUE TITLE
TODO list for 3.1 release on 2.13?

### ISSUE THREAD

**Issue by Daenyth** [scalatest/scalatest#1656]
Is there a todo list for releasing 3.1 on scala 2.13 with scalacheck 1.14?

If there's a public list it would help, so that people can contribute to help speed the release along

**Comment by bvenners** [comment]
Yes, I have a text file. Not much left. I'll be making tickets for them today, but in short:

- Drop `/:` and `\:` from new non-empty anyvals, and deprecate them on `NonEmptyList`, which is the new name for `Chain`. These methods were dropped in 2.13, so we'll follow suit.
- Double check that our API of the NonEmpty's matches the API of 2.13 collections.
- Update our scalafix tool to take care of all the renames we recently added
- Do a pass of all the scaladoc to clean it up, adding in any missing scaladoc
- Possibly add scalactic back into the scalatest scaladoc, because we refer to it in the scalatest scaladoc, and those links were severed when we moved it separate. The idea is we'll still keep a separate scaladoc for scalactic, but also include it in scalatest scaladoc (or figure out some way to link the two).
- The big one: modularize ScalaTest: In 3.1.0 we plan to release the scalatest artifact with deprecated names, so everything will continue to work as before but you'll get deprecation warnings, but we'll also deploy modularized jars that do not include the deprecated aliases. We'll do this for two to three years before removing the deprecated aliases. This enables us to get the benefit of modularization now, which requires that we drop the deprecated names, but keep it easy to upgrade for a very long time to come, which requires that we keep the deprecated names.

That's it, other than we will go through an RC1, RC2, etc., process before we release final. I will be talking about open source in general and ScalaTest 3.1.0 in particular at ScalaWorld Aug 31 to Sep 4, and I would like this to be released final at the very latest by that talk at that conference.

I'll make tickets for some of these, and paste the numbers in here.

**Comment by bvenners** [comment]
Quick update for folks waiting for 3.1.0. Once we add a few more items to the scalafix tool for 3.1.0, we'll release a 3.1.0-M1. Other than documentation (some Scaladoc and release notes) the only major item is modifying the build to modularize the release. Once we have completed the modularization task we'll release 3.1.0-RC1 and start the release candidate (RC) process. Once an RC looks good we'll release 3.1.0 final.
